src/clj/puppetlabs/services/master/file_serving.clj

```diff
@@ -103,16 +190,64 @@
                         project-root
                         (find-project-module project-root module modulepath))
           file-path (str module-root "/" (mount->path-component mount) "/" path)]
-      (if (fs/exists? file-path)
+      (when (fs/exists? file-path)
         file-path))))
 
+(defn mount-dirs-in-modulepath
+  [mount modulepath project-root project-as-module?]
```

Comment:
The two places this is called, the last arg you're passing is `project-name`, which it looks like we extract from the project config. Do we expect that to be nil in some scenarios and that's why it's being treated as a boolean here? I feel like maybe it shouldn't ever be nil, but I may be missing some conflating of ideas that's going on here.

Reply:
Yes, the existence of a project name is how people opt in to treating the project itself as a module. I can try to make this more clear.

Reply:
I abstracted the check into its own function.